internals/secrethub/credential_disable_test.go

```diff
@@ -0,0 +1,141 @@
+		force          bool
+		promptOut      string
+		out            string
+		in             string
+		disableError   error
+		newClientError error
+		promptError    error
+		err            error
+	}{
+		"fail-force-no-credential": {
+			cmd: CredentialDisableCommand{
+				fingerprint: "",
+				force:       true,
+			},
+			err: errors.New("fingerprint argument must be set when using --force"),
+		},
+		"fail-prompt-error": {
+			promptError: testError,
+			err:         testError,
+		},
+		"fail-empty-credential-3-times": {
+			cmd: CredentialDisableCommand{
+				fingerprint: "",
+				force:       false,
+			},
+			promptOut: "What is the fingerprint of the credential you want to disable? \n" +
+				"Invalid input: fingerprint is invalid (api.invalid_fingerprint) \n" +
+				"Please try again.\n" +
+				"What is the fingerprint of the credential you want to disable? \n" +
+				"Invalid input: fingerprint is invalid (api.invalid_fingerprint) \n" +
+				"Please try again.\n" +
+				"What is the fingerprint of the credential you want to disable? \n" +
+				"Invalid input: fingerprint is invalid (api.invalid_fingerprint) \n",
+			err: api.ErrInvalidFingerprint,
+		},
+		"fail-abort": {
+			cmd: CredentialDisableCommand{
+				fingerprint: "62542d734d7f3627",
+				force:       false,
+			},
+			promptOut: "Are you sure you want to disable the credential with fingerprint 62542d734d7f3627? [y/N]: ",
+			in:        "n",
+			out:       "A disabled credential can no longer be used to access SecretHub. This process can currently not be reversed.\nAborting.\n",
+		},
+		"fail-client-error": {
+			newClientError: testError,
+			err:            testError,
+		},
+		"fail-force-wrong-credential": {
+			cmd: CredentialDisableCommand{
+				fingerprint: "BillyBoy",
+				force:       true,
+			},
+			err: api.ErrInvalidFingerprint,
+		},
+		"fail-too-short-fingerprint": {
+			cmd: CredentialDisableCommand{
+				fingerprint: "6254",
+				force:       true,
+			},
+			err: api.ErrTooShortFingerprint,
+		},
+		"succeed-force": {
+			cmd: CredentialDisableCommand{
+				fingerprint: "62542d734d7f3627",
+				force:       true,
+			},
+			out: "A disabled credential can no longer be used to access SecretHub. " +
+				"This process can currently not be reversed.\nCredential disabled.\n",
+		},
+		"succeed-no-force": {
+			cmd: CredentialDisableCommand{
+				fingerprint: "62542d734d7f3627",
+				force:       false,
+			},
+			promptOut: "Are you sure you want to disable the credential with fingerprint 62542d734d7f3627? [y/N]: ",
+			in:        "y",
+			out:       "A disabled credential can no longer be used to access SecretHub. This process can currently not be reversed.\nCredential disabled.\n",
+		},
+		"fail-disable-error": {
+			cmd: CredentialDisableCommand{
+				fingerprint: "62542d734d7f3628",
+				force:       true,
+			},
+			out: "A disabled credential can no longer be used to access SecretHub. " +
+				"This process can currently not be reversed.\n",
```

Comment:
This long string could also be defined as a constant, as it has multiple occurrences.

Reply:
This should now be solved! :D